packages/core/src/state-management/repository.ts

```diff
@@ -0,0 +1,85 @@
+import { Document } from '../document';
+import DocID from '@ceramicnetwork/docid';
+import { AsyncLruMap } from './async-lru-map';
+import { StateStore } from '../store/state-store';
+import { DocumentFactory } from './document-factory';
+import { DocState } from '@ceramicnetwork/common';
+
+export class Repository {
+  readonly #map: AsyncLruMap<Document>;
+  #documentFactory?: DocumentFactory;
+  #stateStore?: StateStore
+
+  constructor(limit: number) {
+    this.#map = new AsyncLruMap(limit, async (entry) => {
```

Comment:
I know you haven't gotten to this yet, but can you help me understand how it will eventually work that if a user has subscribed to updates on a Doctype instance that the corresponding Document/StateManager doesn't get evicted from the cache?